src/main/java/uk/ac/ebi/embl/converter/cli/Main.java

```diff
@@ -164,4 +184,61 @@ public static String getFileExtension(Path path) {
         }
         return null; // No extension found
     }
+
+    static class FormatSupportError extends ExitException {
+        public FormatSupportError(final FileFormat fromFt, final FileFormat toFt) {
+            super("Conversion from \"" + fromFt + "\" to \"" + toFt + "\" is not supported");
+        }
+
+        @Override
+        public CLIExitCode exitCode() {
+            return CLIExitCode.UNSUPPORTED_FORMAT_CONVERSION;
+        }
+    }
+}
+
+record CliRulesOption(Map<ValidationRule, RuleSeverity> rules) {}
+
+class RuleConverter implements ITypeConverter<CliRulesOption> {
+    CliRulesOption map = new CliRulesOption(new HashMap<>());
+
+    @Override
+    public CliRulesOption convert(String args) throws Exception {
+        String[] entries = args.split(",");
+
+        for (String entry : entries) {
+            String[] pairs = entry.trim().split(":");
+            if (pairs.length != 2) {
+                throw new CLIError("Invalid rule: " + entry);
+            }
+            ValidationRule key;
+            try {
+                key = ValidationRule.valueOf(pairs[0].toUpperCase());
+            } catch (IllegalArgumentException e) {
+                throw new CLIError("The rule: \"" + pairs[0] + "\" is invalid");
+            }
+            RuleSeverity value;
+            try {
+                value = RuleSeverity.valueOf(pairs[1].toUpperCase());
+            } catch (IllegalArgumentException e) {
+                throw new CLIError("The rule severity: \"" + pairs[1] + "\" is invalid");
+            }
+            this.map.rules().put(key, value);
+        }
+        return this.map;
+    }
+}
+
+class ExecutionExceptionHandler implements IExecutionExceptionHandler {
+
+    // Tried to use LOG.error instead of println here but would not pipe anything to stderr.
+    @Override
+    public int handleExecutionException(Exception e, CommandLine commandLine, ParseResult parseResult)
+            throws Exception {
+        if (e instanceof ExitException) {
+            System.err.println(e.getMessage());
+            return ((ExitException) e).exitCode().asInt();
+        }
+        throw e;
```

Comment:
We should also catch all other errors and log user-readable messages. We could throw errors in DEBUG mode.

```
if (e instanceof OutOfMemoryError) {
 Log.error("The conversion needs more memory please increase the memory using -Xmx java argument")
}
```